tests/BuildScriptGenerator.Tests/Node/NodePlatformTest.cs

```diff
@@ -0,0 +1,205 @@
+            var repo = new MemorySourceRepo();
+            repo.AddFile(PackageJson, NodeConstants.PackageJsonFileName);
+            var context = CreateContext(repo);
+            context.NodeVersion = "10.10";
+
+            // Act
+            var buildScriptSnippet = nodePlatform.GenerateBashBuildScriptSnippet(context);
+
+            // Assert
+            Assert.NotNull(buildScriptSnippet);
+            Assert.Null(buildScriptSnippet.PlatformInstallationScriptSnippet);
+        }
+
+        [Fact]
+        public void BuildScript_DoesNotHaveSdkInstallScript_IfDynamicInstallNotEnabled_AndSdkIsNotAlreadyInstalled()
+        {
+            // Arrange
+            var nodePlatform = CreateNodePlatform(dynamicInstallIsEnabled: false, sdkAlreadyInstalled: false);
+            var repo = new MemorySourceRepo();
+            repo.AddFile(PackageJson, NodeConstants.PackageJsonFileName);
+            var context = CreateContext(repo);
+            context.NodeVersion = "10.10";
+
+            // Act
+            var buildScriptSnippet = nodePlatform.GenerateBashBuildScriptSnippet(context);
+
+            // Assert
+            Assert.NotNull(buildScriptSnippet);
+            Assert.Null(buildScriptSnippet.PlatformInstallationScriptSnippet);
+        }
+
+        private TestNodePlatform CreateNodePlatform(
+            bool dynamicInstallIsEnabled,
+            bool sdkAlreadyInstalled)
+        {
+            var cliOptions = new BuildScriptGeneratorOptions();
+            cliOptions.EnableDynamicInstall = dynamicInstallIsEnabled;
+            var environment = new TestEnvironment();
+            var installer = new TestNodePlatformInstaller(
+                Options.Create(cliOptions),
+                environment,
+                sdkAlreadyInstalled);
+
+            var versionProvider = new TestNodeVersionProvider();
+            var nodeGeneratorOptions = new NodeScriptGeneratorOptions();
+            var detector = new TestNodeLanguageDetector(
+                versionProvider,
+                Options.Create(nodeGeneratorOptions),
+                NullLogger<NodeLanguageDetector>.Instance,
+                environment,
+                new TestStandardOutputWriter());
+
+            return new TestNodePlatform(
+                Options.Create(cliOptions),
+                Options.Create(nodeGeneratorOptions),
+                versionProvider,
+                NullLogger<NodePlatform>.Instance,
+                detector,
+                environment,
+                installer);
+        }
+
+        private BuildScriptGeneratorContext CreateContext(ISourceRepo sourceRepo)
+        {
+            return new BuildScriptGeneratorContext
+            {
+                SourceRepo = sourceRepo,
+            };
+        }
+
+        private class TestNodePlatform : NodePlatform
+        {
+            public TestNodePlatform(
+                IOptions<BuildScriptGeneratorOptions> commonOptions,
```

Comment:
Maybe it's better to conform the naming to 'cliOptions' same as above? And 'nodeGeneratorOptions' to 'nodeScriptGeneratorOptions'.